File: Largest prime factor[9-August-2023]/Approach 1/Solution.py
#  >> PYTHON CODE <<

import math

class Solution:
    def largestPrimeFactor (self, N):
        def prime(n):
            if(n==2): return True
            for i in range(2,int(math.sqrt(n))+1):
                if(n%i==0): return False
            
            return True
        
        if(prime(N)): return N
        
        ans=-1
        for i in range(2,int(math.sqrt(N))+1):
            if(N%i==0):
                if(prime(i)): ans=max(ans,i)
                if(prime(N/i)): ans=max(ans,N/i)
        
        return int(ans)

File: Largest prime factor[9-August-2023]/Approach 1/solution.cpp
//  >> C++ CODE <<

class Solution{
public: 
    bool prime(int n) {
        if (n == 2) return true;

        for (int i = 2; i <= sqrt(n); i++) {
            if (n % i == 0) return false;
        }

        return true;
    }
    long long int largestPrimeFactor(int N){
        if (prime(N)) return N;

        long ans = -1;

        for (int i = 2; i <= sqrt(N); i++) {
            if (N % i == 0) {
                if (prime(i)) ans = max(ans,long(i));
                if (prime(N / i)) ans = max(ans,long(N / i));
            }
        }

        return ans;
    }
};